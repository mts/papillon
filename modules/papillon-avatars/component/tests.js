import React from 'react';
import {
  avatar,
  avatarSmall,
  avatarParentChild,
  avatarStackMore,
  avatarStackThreePlus
} from './flavors';

describe('<Avatar />', () => {
  let wrapper;

  describe('Instance', () => {
    test('must be an instance of Avatar', () => {
      wrapper = renderMount(avatar);
      expect(wrapper.find('Avatar')).toHaveLength(1);
    });

    test('must be an instance of AvatarParentChild', () => {
      wrapper = renderMount(avatarParentChild);
      expect(wrapper.find('AvatarParentChild')).toHaveLength(1);
    });
  });

  describe('Snaphot', () => {
    test('must match avatar', () => {
      expect(global.renderToJSON(avatar)).toMatchSnapshot();
    });

    test('must match avatar-small', () => {
      expect(global.renderToJSON(avatarSmall)).toMatchSnapshot();
    });

    test('must match avatar-parent-child', () => {
      expect(global.renderToJSON(avatarParentChild)).toMatchSnapshot();
    });

    test('must match avatar-stack-more', () => {
      expect(global.renderToJSON(avatarStackMore)).toMatchSnapshot();
    });

    test('must match avatar-stack-three-plus', () => {
      expect(global.renderToJSON(avatarStackThreePlus)).toMatchSnapshot();
    });
  });
});
