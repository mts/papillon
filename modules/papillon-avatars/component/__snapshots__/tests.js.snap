// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> Snaphot must match avatar 1`] = `
<img
  alt="basic"
  className="avatar"
  height={72}
  src="https://user-images.githubusercontent.com/334891/29999089-2837c968-9009-11e7-92c1-6a7540a594d5.png"
  width={72}
/>
`;

exports[`<Avatar /> Snaphot must match avatar-parent-child 1`] = `
<div
  className="avatar-parent-child float-left"
>
  <img
    alt="parent"
    className="avatar"
    height={48}
    src="https://user-images.githubusercontent.com/334891/29999089-2837c968-9009-11e7-92c1-6a7540a594d5.png"
    width={48}
  />
  <img
    alt="child"
    className="avatar avatar-child"
    height={20}
    src="https://user-images.githubusercontent.com/334891/29999089-2837c968-9009-11e7-92c1-6a7540a594d5.png"
    width={20}
  />
</div>
`;

exports[`<Avatar /> Snaphot must match avatar-small 1`] = `
<img
  alt="small"
  className="avatar avatar-small"
  height={32}
  src="https://user-images.githubusercontent.com/334891/29999089-2837c968-9009-11e7-92c1-6a7540a594d5.png"
  width={32}
/>
`;

exports[`<Avatar /> Snaphot must match avatar-stack-more 1`] = `
<div
  className="AvatarStack AvatarStack--three-plus"
>
  <div
    aria-label="octocat, octocat, octocat, octocat, and octocat"
    className="AvatarStack-body tooltipped tooltipped-se tooltipped-align-left-1"
  >
    <img
      alt="@octocat"
      className="avatar"
      height={20}
      src="https://user-images.githubusercontent.com/334891/29999089-2837c968-9009-11e7-92c1-6a7540a594d5.png"
      width={20}
    />
    <img
      alt="@octocat"
      className="avatar"
      height={20}
      src="https://user-images.githubusercontent.com/334891/29999089-2837c968-9009-11e7-92c1-6a7540a594d5.png"
      width={20}
    />
    <div
      className="avatar avatar-more"
    />
    <img
      alt="@octocat"
      className="avatar"
      height={20}
      src="https://user-images.githubusercontent.com/334891/29999089-2837c968-9009-11e7-92c1-6a7540a594d5.png"
      width={20}
    />
    <img
      alt="@octocat"
      className="avatar"
      height={20}
      src="https://user-images.githubusercontent.com/334891/29999089-2837c968-9009-11e7-92c1-6a7540a594d5.png"
      width={20}
    />
    <img
      alt="@octocat"
      className="avatar"
      height={20}
      src="https://user-images.githubusercontent.com/334891/29999089-2837c968-9009-11e7-92c1-6a7540a594d5.png"
      width={20}
    />
  </div>
</div>
`;

exports[`<Avatar /> Snaphot must match avatar-stack-three-plus 1`] = `
<div
  className="AvatarStack AvatarStack--three-plus"
>
  <div
    aria-label="octocat, octocat, and octocat"
    className="AvatarStack-body tooltipped tooltipped-se tooltipped-align-left-1"
  >
    <img
      alt="@octocat"
      className="avatar"
      height={20}
      src="https://user-images.githubusercontent.com/334891/29999089-2837c968-9009-11e7-92c1-6a7540a594d5.png"
      width={20}
    />
    <img
      alt="@octocat"
      className="avatar"
      height={20}
      src="https://user-images.githubusercontent.com/334891/29999089-2837c968-9009-11e7-92c1-6a7540a594d5.png"
      width={20}
    />
    <img
      alt="@octocat"
      className="avatar"
      height={20}
      src="https://user-images.githubusercontent.com/334891/29999089-2837c968-9009-11e7-92c1-6a7540a594d5.png"
      width={20}
    />
  </div>
</div>
`;
